web/src/settings_profile_fields.js

```diff
@@ -147,22 +147,26 @@ function read_external_account_field_data($profile_field_form) {
 export function update_choice_delete_btn($container, display_flag) {
     const no_of_choice_row = $container.find(".choice-row").length;
 
-    // Disable delete button if there only one choice row
-    // Enable choice delete button more one than once choice
-    if (no_of_choice_row === 1) {
+    // Comparing with 2 because including extra empty choice row.
+    if (no_of_choice_row === 2) {
         if (display_flag === true) {
-            $container.find(".choice-row .delete-choice").show();
+            $container.find(".choice-row .delete-choice").first().show();
         } else {
-            $container.find(".choice-row .delete-choice").hide();
+            $container.find(".choice-row .delete-choice").first().hide();
         }
     }
 }
 
 export function get_value_for_new_option(container) {
+    const $choice_rows = $(container).find(".choice-row");
     let value = 0;
-    for (const row of $(container).find(".choice-row")) {
+    for (const row of $choice_rows) {
         value = Math.max(value, Number.parseInt($(row).attr("data-value"), 10) + 1);
     }
+
+    // Display delete button if choice have some value.
+    $choice_rows.find(".delete-choice").removeClass("hide");
```

Comment:
I don't think this is the correct function to have this code for showing/hiding the choice.

To be clear, the "value" here is to set the `data-value` attribute of the choice element and not the value inside the input.

I think a more cleaner way to handle the display of delete buttons is to just hide the button for the new row (which I guess is handled using `create_choice_row` and `add_choice_row` functions) and then show the button for the row above the newly added row.